### ISSUE TITLE
Trouble with labels showing up

### ISSUE THREAD

**Issue by shepherdjay** [samilliken/openDCIM#1157]
Running version 19.01 - We have a handful of devices that are all using a Cisco ASR-1001 template. They are so far the only devices we have where the front label doesn't show up.

Example here: 
![image](https://user-images.githubusercontent.com/13685721/59229629-30b57900-8b98-11e9-909e-dde788bdb020.png)

If images are turned off it still doesn't show up:
![image](https://user-images.githubusercontent.com/13685721/59229665-4460df80-8b98-11e9-9014-c9942c12f078.png)

If I remove the front image from the template it does show a label but obviously no picture. Not sure what is going on here. I've tried cloning template and problem still exists. Working on recreating it from scratch and can report back but that can be tedious if we end up having to do that down the line with a lot of the same type.

**Comment by wilpig** [comment]
Recreate this problem on dev.opendcim.org and I'll look into it

On Jun 10, 2019 5:57 PM, Jay Shepherd <notifications@github.com> wrote:

Running version 19.01 - We have a handful of devices that are all using a Cisco ASR-1001 template. They are so far the only devices we have where the front label doesn't show up.

Example here:
[image]<https://user-images.githubusercontent.com/13685721/59229629-30b57900-8b98-11e9-909e-dde788bdb020.png>

If images are turned off it still doesn't show up:
[image]<https://user-images.githubusercontent.com/13685721/59229665-4460df80-8b98-11e9-9014-c9942c12f078.png>

If I remove the front image from the template it does show a label but obviously no picture. Not sure what is going on here. I've tried cloning template and problem still exists. Working on recreating it from scratch and can report back but that can be tedious if we end up having to do that down the line with a lot of the same type.

**Comment by shepherdjay** [comment]
Put it on here:
https://dev.opendcim.org/cabnavigator.php?cabinetid=80

As well as here:
https://opendcim.org/demo/cabnavigator.php?cabinetid=5

**Comment by wilpig** [comment]
Definite bug, I will be home from vacation tomorrow and i'll dig into it further.

https://dev.opendcim.org/api/v1/device/5158/getpicture
https://dev.opendcim.org/api/v1/device/5158/getpicture?rear

In looking at the output from the api it didn't generate the code for the label.  So i'm gonna have to debug the function that builds those to figure out why it didn't generate the label.

**Comment by wilpig** [comment]
Scratch that, not a bug.  Not fully working as you want but I have a reason and it's documented!

https://github.com/samilliken/openDCIM/blob/master/classes/Device.class.php#L2169

**Comment by shepherdjay** [comment]
Yeah I did consider it having to do with the chassis and had switched the device to a switch with no change. Perhaps because I didn’t zero out the slot count?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by wilpig** [comment]
Almost certainly.  I don't think we zero out the slot counts when devices get their type switched and we should probably put that in the device update function to keep the data sane

**Comment by wilpig** [comment]
https://github.com/samilliken/openDCIM/blob/master/classes/Device.class.php#L748

We kill any children if you switch the device type, we should zero out the slot definitions at the same time.

**Comment by shepherdjay** [comment]
It looks like as well if the template switches types and you click update devices the devices don't actually update to the same type.

This might be by design? It explains why when switching template your earlier line about not drawing label still applied. I hadn't actually updated child device types.

**Comment by shepherdjay** [comment]
> This might be by design? It explains why when switching template your earlier line about not drawing label still applied. I hadn't actually updated child device types.

Confirmed by design https://github.com/samilliken/openDCIM/blob/master/classes/DeviceTemplate.class.php#L410

Would it make sense to alter the chassis label logic to check that at least one slot is filled, in addition to your 3u criteria, instead of checking strictly for slots > 0

Or perhaps just give the user a notice someway when configuring chassis type device/template and height < 3

I think hiding the label was a reasonable decision in hindsight but it isn't very clear, at least to me, what the user experience would be. Granted setting a router as a chassis maybe not ideal but was unsure best way to handle the expansion bay on them.
